adaptive/src/main/java/com/google/accompanist/adaptive/TwoPane.kt

```diff
@@ -0,0 +1,545 @@
+)
+
+/**
+ * A strategy for configuring the [TwoPane] component, that is responsible for the meta-data
+ * corresponding to the arrangement of the two panes of the layout.
+ */
+public fun interface TwoPaneStrategy {
+    /**
+     * Calculates the split result in local bounds of the [TwoPane].
+     *
+     * @param density the [Density] for measuring and laying out
+     * @param layoutDirection the [LayoutDirection] for measuring and laying out
+     * @param layoutCoordinates the [LayoutCoordinates] of the [TwoPane]
+     */
+    public fun calculateSplitResult(
+        density: Density,
+        layoutDirection: LayoutDirection,
+        layoutCoordinates: LayoutCoordinates
+    ): SplitResult
+}
+
+/**
+ * A strategy for configuring the [TwoPane] component, that is responsible for the meta-data
+ * corresponding to the arrangement of the two panes of the layout.
+ *
+ * This strategy can be conditional: If `null` is returned from [calculateSplitResult], then this
+ * strategy did not produce a split result to use, and a different strategy should be used.
+ */
+private fun interface ConditionalTwoPaneStrategy {
+    /**
+     * Calculates the split result in local bounds of the [TwoPane], or `null` if this strategy
+     * does not apply.
+     *
+     * @param density the [Density] for measuring and laying out
+     * @param layoutDirection the [LayoutDirection] for measuring and laying out
+     * @param layoutCoordinates the [LayoutCoordinates] of the [TwoPane]
+     */
+    public fun calculateSplitResult(
+        density: Density,
+        layoutDirection: LayoutDirection,
+        layoutCoordinates: LayoutCoordinates
+    ): SplitResult?
+}
+
+/**
+ * Returns a [TwoPaneStrategy] that will place the slots vertically or horizontally if there is a
+ * horizontal or vertical fold respectively.
+ *
+ * If there is no fold, then the [defaultStrategy] will be used instead.
+ */
+public fun TwoPaneStrategy(
+    windowGeometry: WindowGeometry,
+    defaultStrategy: TwoPaneStrategy,
+): TwoPaneStrategy = TwoPaneStrategy(
+    FoldAwareHorizontalTwoPaneStrategy(windowGeometry),
+    FoldAwareVerticalTwoPaneStrategy(windowGeometry),
+    defaultStrategy = defaultStrategy
+)
+
+/**
+ * Returns a [TwoPaneStrategy] that will place the slots horizontally if there is a vertical fold.
+ *
+ * If there is no fold, then the [defaultStrategy] will be used instead.
+ */
+public fun HorizontalTwoPaneStrategy(
+    windowGeometry: WindowGeometry,
+    defaultStrategy: TwoPaneStrategy,
+): TwoPaneStrategy = TwoPaneStrategy(
+    FoldAwareHorizontalTwoPaneStrategy(windowGeometry),
+    defaultStrategy = defaultStrategy
+)
+
+/**
+ * Returns a [TwoPaneStrategy] that will place the slots horizontally.
+ *
+ * If there is a vertical fold, then the gap will be placed along the fold.
+ *
+ * Otherwise, the gap will be placed at the given [splitFraction] from start, with the given
+ * [gapWidth].
+ */
+public fun HorizontalTwoPaneStrategy(
```

Comment:
You can probably half the number of APIs if you provide an orientation: Orientation parameter.

Reply:
I think I'd prefer to keep it is as-is. Having the name split means the parameter names can be orientation specific (`gapWidth` vs `gapHeight`) and clearer behavior when it comes to LTR and RTL.

Reply:
ack, let's roll with it!